### PR TITLE
ATA-5400 Move generator from prism-sdk to this repo

### PR DESCRIPTION
**Pull request by pva701**
This PR moves grpc clients generator to the pbandk repo.
 Related PR is [here](https://github.com/input-output-hk/atala-prism-sdk/pull/143)

### CODE REVIEW

File: prism-clients-generator/build.gradle.kts
Comment (AlexITC) [inline prism-clients-generator/build.gradle.kts:7]: This is not a PRISM client generator anymore but a gRPC client generator, I'd try to keep this generic with the goal to get it merged to the upstream project (https://github.com/streem/pbandk).

**Comment by AlexITC** [comment]
@pva701 please make sure the branch is up to date

File: prism-clients-generator/build.gradle.kts
Comment (pva701) [reply prism-clients-generator/build.gradle.kts:7]: @AlexITC I am totally agree, and I called the project `grpc-clients-generator` at first but there is a tiny issue why I decided to use `prism` in the name and everywhere, it's `GrpcClient` class. I mean those generated clients are relying on the `GrpcClient` which has methods with `PrismMetadata`.

Also all generated clients have methods which accept `PrismMetadata`, so I concluded that for now the generator is tied with `prism-protos` `GrpcClient`.  Should we do something about this?
Also, those auth and not auth methods seem to be prism-specific.

For now I would leave it as it's and later we can think if it's possible to make the generator general to merge to upstream. what do you think?

Comment (vlad107) [reply prism-clients-generator/build.gradle.kts:7]: yeah, I agree with Ilya -- Dan started two discussions about potentially useful features for upstream: https://github.com/streem/pbandk/discussions/173 & https://github.com/streem/pbandk/discussions/174

but this one looks more PRISM specific  and it's not obvious how to make it useful for someone else

Comment (AlexITC) [reply prism-clients-generator/build.gradle.kts:7]: @pva701 thanks for the explanation, I'd do this:
- Get this merged to unblock the work.
- Create a task for refactoring this generator in a way that it isn't prism-specific but generic enough to allow including the prism-specific functionality when depending on the generator.
- Get the dependent PR merged too.

**Review by AlexITC — APPROVED** [review]

**Comment by pva701** [comment]
@vlad107 plz wait and don't merge the PR, I figured out the small issue that a package of generated classes is `io.iohk.atala.prism.protos` but in the vault it should be `io.iohk.atala.vault.protos` or `io.iohk.atala.prism.vault.protos` I guess, so it's better make customisable. I will check how easy it to achieve.

**Pull request merged**
